webapp/Sigbuilder-graph.js

```diff
@@ -0,0 +1,1872 @@
+    myform.appendChild(linebreak);
+
+    var labelArray = ['Filename','Writing [C] format'];
+    var textValueArray = ["mydatafile.dat","%g %g"];
+    for(var i = 0; i < labelArray.length; i++){
+        // Input Title
+        var namelabel = document.createElement('label');
+        namelabel.innerHTML = labelArray[i];
+        namelabel.style.marginLeft = "30px";
+        myform.appendChild(namelabel);
+
+        // Input
+        var input = document.createElement("input");
+        input.name = "edit_"+labelArray[i].toString();
+        input.value = textValueArray[i];
+        input.setAttribute("id", "filesave_"+i.toString());
+        input.setAttribute("class", "fieldInput");
+        myform.appendChild(input);
+
+        // Line break
+        var linebreak = document.createElement('br');
+        myform.appendChild(linebreak);
+
+        // Line break
+        var linebreak = document.createElement('br');
+        myform.appendChild(linebreak);
+
+    }
+    // Line break
+    var linebreak = document.createElement('br');
+    myform.appendChild(linebreak);
+
+    // Button - Cancel
+    var cancel_btn = document.createElement("button");
+    cancel_btn.style.cssFloat = "right";
+    cancel_btn.innerHTML = 'Cancel';
+    cancel_btn.type = "button";
+    cancel_btn.name = "Cancel";
+    myform.appendChild(cancel_btn);
+
+    // Button - OK
+    var ok_btn = document.createElement("button");
+    ok_btn.style.cssFloat = "right";
+    ok_btn.style.marginRight = "20px";
+    ok_btn.innerHTML = 'OK';
+    ok_btn.type = "button";
+    ok_btn.name = "OK";
+
+    myform.appendChild(ok_btn);
+    content.appendChild(myform);
+    var height = 180;
+    var wind = showModalWindow(graph, 'Scilab Multiple Values Request', content, 300, height);
+
+    // Executes when button 'cancel_btn' is clicked
+    cancel_btn.onclick = function() {
+        document.getElementById("messageLabel").innerHTML = "";
+        wind.destroy();
+    };
+    // Executes when button 'ok_btn' is clicked
+    ok_btn.onclick = function() {
+        var pointlength = sigbuilder_Graph.series[0].data.length;
+        var format = (document.getElementById("filesave_1").value).trim();
+        var count = (format.match(/%/g) || []).length;
+        if(count == 2){
+            var firstchar = format.charAt((format.indexOf("%"))+1);
+            var secondchar = format.charAt((format.lastIndexOf("%"))+1);
+            var ary = ["%"+firstchar,"%"+secondchar];
+            if((firstchar == 'g'||firstchar == 'f')&&(secondchar == 'g'||secondchar == 'f')){
+            if ("Blob" in window) {
+                var fileName = document.getElementById("filesave_0").value;
+                var textToWrite = "";
+                var x = 0;
+                var y = 0;
+                for (var i=0;i<pointlength;i++) {
+                    if(ary[0] == "%g"){
+                        if((sigbuilder_Graph.series[0].data[i].x).toString().includes(".")){
```

Comment:
```suggestion
                        var varx = sigbuilder_Graph.series[0].data[i].x;
                        if (varx.toString().includes(".")) {
```
line is too long. add a var first and use it everywhere.